src/main/kotlin/org/ageseries/libage/sim/thermal/Constructs.kt

```diff
@@ -0,0 +1,124 @@
+package org.ageseries.libage.sim.thermal
+
+import org.ageseries.libage.sim.constant.Material
+import org.ageseries.libage.sim.measurement.Scale
+import org.ageseries.libage.sim.measurement.ThermalUnits
+import kotlin.math.pow
+
+/**
+ * A temperature.
+ *
+ * The inner unit type is always Kelvin; conversions are available as functions and properties.
+ */
+@JvmInline
+value class Temperature(val kelvin: Double) {
+    /**
+     * Return this temperature on the given [Scale].
+     */
+    fun to(scale: Scale): Double = scale.map(kelvin)
+
+    val absoluteGrade get() = to(ThermalUnits.ABSOLUTE_GRADE)
+    val rankine get() = to(ThermalUnits.RANKINE)
+    val grade get() = to(ThermalUnits.GRADE)
+    val centigrade get() = to(ThermalUnits.CENTIGRADE)
+    val celsius get() = to(ThermalUnits.CELSIUS)
+    val milligrade get() = to(ThermalUnits.MILLIGRADE)
+    val fahrenheit get() = to(ThermalUnits.FAHRENHEIT)
+
+    operator fun plus(rhs: Temperature) = Temperature(kelvin + rhs.kelvin)
+    operator fun minus(rhs: Temperature) = Temperature(kelvin - rhs.kelvin)
+
+    // Would it make sense to multiply 2 temps?
+    operator fun times(rhs: Double) = Temperature(kelvin * rhs)
+    operator fun div(rhs: Double) = Temperature(kelvin / rhs)
+    operator fun compareTo(rhs: Temperature) = kelvin.compareTo(rhs.kelvin)
+    override fun toString() = ThermalUnits.KELVIN.display(kelvin)
+
+    companion object {
+        /**
+         * Return a Temperature given a measurement on a different [Scale].
+         */
+        fun from(temp: Double, scale: Scale) = Temperature(scale.unmap(temp))
+    }
+}
+
+/**
+ * "Standard" temperature, in Kelvin.
+ *
+ * This is the "ST" of STP, as reported in laboratory conditions.
+ *
+ * This is used as the default temperature for thermal masses.
+ */
+val STANDARD_TEMPERATURE: Temperature = Temperature(273.15)
+
+/**
+ * Represents a mass with thermal energy.
+ * */
+class ThermalMass(
+    /** The material of this mass, used for its thermal properties. */
+    val material: Material,
+    /** Thermal energy, in J. Leave null to set [STANDARD_TEMPERATURE]. */
+    energy: Double? = null,
+    /** Mass, in kg. */
+    val mass: Double = 1.0
+) {
+
+    var energy: Double = energy ?: (STANDARD_TEMPERATURE.kelvin * mass * material.specificHeat)
+
+    /**
+     * Temperature of this mass, in K.
+     *
+     * Setting this changes the [energy].
+     */
+    var temperature: Temperature
+        get() = Temperature(energy / mass / material.specificHeat)
+        set(value) {
+            energy = value.kelvin * mass * material.specificHeat
+        }
+
+    override fun toString() = "<Thermal Mass $material ${mass}kg ${energy}J $temperature>"
+}
+
+data class ThermalConnectionParameters(
```

Comment:
Might want to use `open` here if we want to extend it with a serializable version.